core/control-plane/contract/src/main/java/org/eclipse/dataspaceconnector/contract/offer/ContractOfferServiceImpl.java

```diff
@@ -57,8 +60,19 @@ public Stream<ContractOffer> queryContractOffers(ContractOfferQuery query, Range
         var agent = agentService.createFor(query.getClaimToken());
 
         return definitionService.definitionsFor(agent, range)
+                // ignore definitions with missing criteria (only if not SELECT_ALL)
+                .filter(definition -> (definition.getSelectorExpression().getCriteria() != null &&
+                        !definition.getSelectorExpression().getCriteria().isEmpty()) ||
+                        definition.getSelectorExpression() == AssetSelectorExpression.SELECT_ALL)
                 .flatMap(definition -> {
-                    var assets = assetIndex.queryAssets(definition.getSelectorExpression());
+                    Stream<Asset> assets;
+                    if (definition.getSelectorExpression() == AssetSelectorExpression.SELECT_ALL) {
```

Comment:
Pay attention on using `==` to equal on object instances, however I see no use for this `if` block

Reply:
This if block adds the logic that `SELECT_ALL` overwrites any filters as it applies a `ContractDefinition` to any `Asset`. As above, the methods adds logic that the `AssetSelectorExpression` does not provide itself.